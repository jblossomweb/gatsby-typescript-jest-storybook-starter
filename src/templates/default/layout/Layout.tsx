import React from "react"
import PropTypes from "prop-types"

import Header from "../header"
import Footer from "../footer"
import "./Layout.style.css"

export interface Props {
  children: React.ReactNode
  title?: string
}

const Layout: React.FC<Props> = ({ children, title }) => (
  <>
    <Header siteTitle={title} />
    <div
      style={{
        margin: `0 auto`,
        maxWidth: 960,
        padding: `0 1.0875rem 1.45rem`,
      }}
    >
      <main>{children}</main>
      <Footer />
    </div>
  </>
)

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  title: PropTypes.string,
}

export default Layout
