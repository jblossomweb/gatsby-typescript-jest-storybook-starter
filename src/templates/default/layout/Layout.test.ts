import snapshotTests from "../../../../utils/story-snapshot-tests"
import { stories } from "./Layout.stories"

describe("templates/default/layout", () => {
  snapshotTests(stories)
})
