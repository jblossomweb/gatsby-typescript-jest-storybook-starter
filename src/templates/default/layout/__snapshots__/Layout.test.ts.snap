// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates/default/layout no title matches snapshot 1`] = `
Array [
  <header
    style={
      Object {
        "background": "rebeccapurple",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          
        </a>
      </h1>
    </div>
  </header>,
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0 1.0875rem 1.45rem",
      }
    }
  >
    <main>
      Sample Children
    </main>
    <footer>
      © 
      2020
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
        target="_blank"
      >
        Gatsby
      </a>
    </footer>
  </div>,
]
`;

exports[`templates/default/layout sample matches snapshot 1`] = `
Array [
  <header
    style={
      Object {
        "background": "rebeccapurple",
        "marginBottom": "1.45rem",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 960,
          "padding": "1.45rem 1.0875rem",
        }
      }
    >
      <h1
        style={
          Object {
            "margin": 0,
          }
        }
      >
        <a
          href="/"
          style={
            Object {
              "color": "white",
              "textDecoration": "none",
            }
          }
        >
          Sample Title
        </a>
      </h1>
    </div>
  </header>,
  <div
    style={
      Object {
        "margin": "0 auto",
        "maxWidth": 960,
        "padding": "0 1.0875rem 1.45rem",
      }
    }
  >
    <main>
      Sample Children
    </main>
    <footer>
      © 
      2020
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
        target="_blank"
      >
        Gatsby
      </a>
    </footer>
  </div>,
]
`;
